phost/src/main.rs

```diff
@@ -503,17 +524,39 @@ async fn batch_sync_block(
         let r = req_sync_header(pr, &header_batch, authrotiy_change.as_ref()).await?;
         println!("  ..sync_header: {:?}", r);
 
-        let para_fin_header_data = chain_client::get_parachain_heads(
+        let raw_header = match chain_client::get_storage(
             &client,
             Some(last_header_hash),
-            paraid_storage_key.clone(),
-        )
-        .await?;
+            para_head_storage_key.clone())
+            .await? {
+            Some(head) => head,
+            None => {
+                return Err(Error::FailedToDecode);
+            }
+        };
```

Comment:
```suggestion
            .await?.ok_or(Error::EmptyParaHead)?;
```

Let's add `Error::EmptyParaHead` to Error